### PR TITLE
fix(5.1.0): type check for spatial and nested values

### PR DESCRIPTION
**Pull request by Bento007**
## Reason for Change

- #862

## Changes

- assert the `uns['spatial]`, `uns['spatial'][library_id]["scalefactors"]`, and `uns['spatial'][library_id]["images"]` or type `dict`.

## Testing

- updated tests cases with new error messages and added additional test cases to verify new functionality.

## Notes for Reviewer

### CODE REVIEW

**Review by nayib-jose-gloria — APPROVED** [review]

**Comment by nayib-jose-gloria** [comment]
@brian-mott ready for your review!

**Review by MillenniumFalconMechanic — APPROVED** [review]

**Comment by brian-mott** [comment]
If I set `uns['spatial']` to the value of `None`, `True`, `False`, `1`, or `1.0`, I get this error:
`TypeError: argument of type '{type(value)}' is not iterable`
at [this line](https://github.com/chanzuckerberg/single-cell-curation/blob/068921aea51e0a0525b6747a2b075adc95d73da1/cellxgene_schema_cli/cellxgene_schema/validate.py#L78)

**Comment by Bento007** [comment]
@brian-mott that case is resolved

File: cellxgene_schema_cli/cellxgene_schema/validate.py [hunk 1]
```diff
@@ -79,7 +79,10 @@ def _is_single(self) -> bool | None:
         """
         return (
             self.adata.uns["spatial"]["is_single"]
-            if hasattr(self.adata, "uns") and "spatial" in self.adata.uns and "is_single" in self.adata.uns["spatial"]
+            if hasattr(self.adata, "uns")
+            and "spatial" in self.adata.uns
+            and self.adata.uns["spatial"]
+            and "is_single" in self.adata.uns["spatial"]
```
Comment (brian-mott) [inline cellxgene_schema_cli/cellxgene_schema/validate.py:85]: @Bento007 `None` now passes 
But setting `uns['spatial']` = to `1`, `1.0`, or `True` raises this TypeError at this line: `TypeError: argument of type '{ int | float | bool}' is not iterable`

File: cellxgene_schema_cli/cellxgene_schema/validate.py [hunk 2]
```diff
@@ -79,7 +79,10 @@ def _is_single(self) -> bool | None:
         """
         return (
             self.adata.uns["spatial"]["is_single"]
-            if hasattr(self.adata, "uns") and "spatial" in self.adata.uns and "is_single" in self.adata.uns["spatial"]
+            if hasattr(self.adata, "uns")
+            and "spatial" in self.adata.uns
```
Comment (brian-mott) [inline cellxgene_schema_cli/cellxgene_schema/validate.py:83]: If I set `uns['spatial']` = pandas dataframe, I get this exception at this line:
`ValueError: The truth value of a DataFrame is ambiguous. Use a.empty, a.bool(), a.item(), a.any() or a.all().`

**Review by brian-mott — APPROVED** [review]
This looks good for this issue.
Just want to remind about [this comment](https://github.com/chanzuckerberg/single-cell-curation/issues/862#issuecomment-2083407491) so it is not forgotten

**Comment by Bento007** [comment]
> This looks good for this issue. Just want to remind about [this comment](https://github.com/chanzuckerberg/single-cell-curation/issues/862#issuecomment-2083407491) so it is not forgotten

Those cases are also covered by this PR

**Pull request merged**